src/parse/matches/arg_matches.rs

```diff
@@ -416,20 +420,24 @@ impl ArgMatches {
     pub fn values_of_t<R>(&self, name: &str) -> Result<Vec<R>, Error>
     where
         R: FromStr,
-        <R as FromStr>::Err: Display,
+        <R as FromStr>::Err: Display + Debug,
     {
         if let Some(vals) = self.values_of(name) {
             vals.map(|v| {
-                v.parse::<R>().map_err(|e| {
-                    Error::value_validation_auto(&format!(
+                let parsed = v.parse::<R>();
+
+                if let Err(e) = parsed {
+                    Err(Error::value_validation_auto(&format!(
                         "The argument '{}' isn't a valid value: {}",
                         v, e
-                    ))
-                })
+                    ))?)
+                } else {
+                    Ok(parsed.expect(INTERNAL_ERROR_MSG))
+                }
             })
             .collect()
         } else {
-            Err(Error::argument_not_found_auto(name))
+            Err(Error::argument_not_found_auto(name)?)
```

Comment:
Come to think about it, we should probably return an empty `Vec` here. Number of values can be zero after all.